client.go

```diff
@@ -150,17 +145,8 @@ func newClient() (*client, error) {
 
 // Close is used to cleanup the client
 func (c *client) Close() error {
-	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
-	if c.closed {
-		return nil
-	}
-	c.closed = true
-
-	log.Printf("[INFO] mdns: Closing client %v", *c)
+	log.Printf("[INFO] mdns: Closing client: %v", *c)
 	close(c.closedCh)
```

Comment:
@asim's proposed solution is racy. If you want to provide reentrancy, I think #54 is better. I would also say that this PR comes up with a more complicated two-layer select loop in `query()`, while #54 keeps it easier.